pyvisa_py/protocols/hislip.py

```diff
@@ -0,0 +1,714 @@
+"""
+    Python implementation of HiSLIP protocol.  Based on the HiSLIP spec:
+
+    http://www.ivifoundation.org/downloads/Class%20Specifications/IVI-6.1_HiSLIP-1.1-2011-02-24.pdf
+"""
+
+import socket
+import struct
+import time
+from typing import Dict, Union
+
+PORT = 4880
+
+LookupTable = Dict[Union[int, str], Union[int, str]]
+
+MESSAGETYPE: LookupTable = {
```

Comment:
What advantage to do you see in merging the forward (key-> value) and backward (value->key) mapping. It looks to me like it is always clear which one should be used. We could keep them separate and get stricter typing.

Reply:
For me the only advantage is that I don't have to think of two good, unique names.  Naming things is hard, and I'm lazy.  Think of it as a form of polymorphism.

Reply:
Fair enough